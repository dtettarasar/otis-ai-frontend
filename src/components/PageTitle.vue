<script>

    export default {
        name: 'PageTitle',
        props: {
            txtValue: {
            type: String,
            required: true
            }
        }
    }

</script>
<template>
    <div>
        <h1 id="page-title">{{ txtValue }}</h1>
    </div>
</template>
<style>

</style>