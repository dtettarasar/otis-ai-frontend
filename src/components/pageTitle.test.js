import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import PageTitle from './PageTitle.vue'

describe('PageTitle.vue', () => {

  it('renders props.txtValue when passed', () => {
    const txtValue = 'Test Title';
    const wrapper = mount(PageTitle, {
      props: { txtValue }
    });
    expect(wrapper.find('#page-title').text()).toBe(txtValue);
  })

  it('matches the snapshot', () => {
    const txtValue = 'Snapshot Title'
    const wrapper = mount(PageTitle, {
      props: { txtValue }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  
})
