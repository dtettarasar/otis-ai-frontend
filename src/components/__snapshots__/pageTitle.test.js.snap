// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle.vue > matches the snapshot 1`] = `
"<div>
  <h1 id="page-title">Snapshot Title</h1>
</div>"
`;
